### DISCUSSION TITLE
git-bash can not load `$HOME` configure

### DISCUSSION THREAD

**Discussion by moreant** [Zhengqbbb/cz-git#125] [Q&A]
发现在 ~/.czrc 里配置的 useEmoji 和 types 不生效，但是在本地目录的或者使用 --config 指定的却又生效。

![image](https://user-images.githubusercontent.com/45537528/236596111-54aef7d9-6fdd-40c8-902b-292007cfd6e0.png)

经过测试与查看源码，发现实际生效的只有 `openAIToken` 与 `apiProxy` 两个配置，请问这是 feature 还是 bug？
我的环境是 Windows，但是配置的 openAIToken 是可以生效的，排除掉没读取到正确文件的问题。

![image](https://user-images.githubusercontent.com/45537528/236596317-35b8ea4f-bf2e-4155-a0e9-4afd825dd2ce.png)

https://github.com/Zhengqbbb/cz-git/blob/1726b877b080c0909fd94299c292b7fbc5726f29/packages/%40cz-git/plugin-loader/src/index.ts#L134-L150

**Comment by Zhengqbbb** [comment]
rootLoader  优先 项目本地配置 再到 root-config

之所以不生效我猜测，你项目 `package.json` 含有 `config.commitizen` 或是项目根目录中含有 `.czrc` 文件

解决方案：
1. 编写配置于 项目之中
2. 编写全局 commitlint 配置，如：`~/.commitlintrc.cjs`

---

这个函数主要是加载 `openAIToken`   `apiProxy` 这两个配置，加载 `~/.czrc` 是为了兼容旧版本。
因为这涉及到 apikey 的隐私配置，要单分一个加载函数以及文件来进行保存，目的就是为了隔离普通的配置文件，提供给敏感用户可为此文件添加权限进行读写允许
所以目前推荐写入到 `~/.config/.czrc` 当中比较好。推荐使用配置命令行选项进行配置
非常感谢提醒，后面函数名还是进行更改比较好 => `aiConfigLoader`

**Comment by moreant** [comment]
可是我的项目是新建的，没有在 `package.json` 里配置什么，使用 https://github.com/Zhengqbbb/cz-git/issues/42#issuecomment-1168390553 中提到的测方法测试后的结果是这样的：

![image](https://user-images.githubusercontent.com/45537528/237014214-be9f20df-ce82-40cf-93ac-37dbe81582e8.png)

然后我在 cgz 编译后的 `index.js` 文件中的 `rootLoader` 方法中添加了 `...e` 之后就能正常读取到 ~/.czrc 的配置了。
![image](https://user-images.githubusercontent.com/45537528/237015526-51f18bba-6064-4d7b-89e9-32ebd2a0a79d.png)

**Reply by Zhengqbbb** [reply]
我怀疑是因为 commit-loader 和 commitlint-loader 并没有正确读取到，在 git-bash环境中，我做出了下列更改，并发布了 next 版本。
因为我目前手里没有 windows 电脑，希望你可以协助进行测试，**感激不尽** 🙏。

https://github.com/Zhengqbbb/cz-git/commit/44dd794b531a9a76e4dd2667540dd2e75a758df5

```sh
npm install -g czg@1.6.2-beta.5
```

**Reply by moreant** [reply]
客气了，是我感谢您对 Windows 用户的支持才对。🤣

--- 

试了一下 1.6.2-beta.5 还是没能读取到配置文件的路径。我将 loader 方法的入参和几个参数都输出出来了，看了一下像是 `cosmiconfig` 没能正确读到配置文件。我没用过这个包，还得您先看看是怎么回事了。

https://github.com/Zhengqbbb/cz-git/blob/44dd794b531a9a76e4dd2667540dd2e75a758df5/packages/%40cz-git/plugin-loader/src/index.ts#L25-L41

这个是我在 return 前加的输出，下方图片是输出内容。
```js
console.log({ options, resultPath, result })
```

![Snipaste_2023-05-11_07-19-40](https://github.com/Zhengqbbb/cz-git/assets/45537528/3cc2b8eb-1a88-43f4-a197-441e4f398ef1)

**Reply by Zhengqbbb** [reply]
看样子确实有问题，不过我还想知道如果 当前目录中有配置文件是否会正常加载

---

目前来看解决方案只有 alias 了
```sh
echo 'alias cz="czg --config=/c/Users/molingran/.czrc"' >> ~/.bashrc
bash
cz
```

**Reply by moreant** [reply]
当前目录有配置文件的话会被加载的。

---

哈哈，目前我就是这样解决的。alias 是 `czgc` 表示 czg with root config 。使用起来没啥问题。

**Comment by Zhengqbbb** [comment]
czg 🌟(v1.7.0)🌟 版本已经发布，其中修复了在 Windows 系统中 `~/.czrc` 配置不生效的问题 #132 

可以进行更新测试，希望可以帮助到你～ 如果还有问题可以及时和我沟通 🤗

另外，这个版本我还添加了配置文件的路径输出，提供调试
```
CZ_DEBUG=1 czg
```
